### PR TITLE
Swapd: Use transaction directly from transaction confirmations

### PR DESCRIPTION
**Pull request by TheCharlatan**
When receiving a transaction containing adaptor keys, extract them immediately and do not wait for finality.

### CODE REVIEW

File: src/swapd/swap_state.rs [hunk 1]
```diff
@@ -1854,80 +1794,86 @@ fn try_alice_canceled_to_alice_refund_or_alice_punish(
...
+                    )?;
+
+                    runtime
+                        .syncer_state
+                        .tasks
+                        .txids
+                        .remove_entry(&TxLabel::Refund);
+
+                    // Check if we already registered the lock transaction, if so, initiate sweeping procedure
+                    runtime.log_debug(format!("{:?}", runtime.syncer_state.confirmations));
```
Comment (h4sh3d) [inline src/swapd/swap_state.rs:1826]: This part didn't change but is it necessary to have `{:?}` fmt?

File: src/swapd/swap_state.rs [hunk 2]
```diff
@@ -108,7 +105,7 @@ use super::{
 ///       |_______________________|                                   |                AliceCancel
 ///       |                       |                                   |                     |
 ///       |                       V                                   |         ____________|
-///       V                   BobBuyFinal                             |        |            |
+///       V                   BobBuySeen                             |        |            |
```
Comment (h4sh3d) [inline src/swapd/swap_state.rs:108]: Also `s/try_bob_buy_final_to_bob_buy_sweeping/try_bob_buy_seen_to_bob_buy_sweeping/`

**Review by h4sh3d — CHANGES_REQUESTED** [review]
cACK

Maybe we want this to run on farcaster.dev for a bit?

**Comment by TheCharlatan** [comment]
> cACK
> 
> Maybe we want this to run on farcaster.dev for a bit?

Yes!

File: src/swapd/swap_state.rs (hunk 1 above)
Comment (TheCharlatan) [reply src/swapd/swap_state.rs:1826]: It's a HashMap, not sure how else to handle this.

File: src/swapd/swap_state.rs (hunk 2 above)
Comment (TheCharlatan) [reply src/swapd/swap_state.rs:108]: ce0669d326d0cadb58996117a32061f217cee082

File: src/swapd/swap_state.rs [hunk 3]
```diff
@@ -1170,68 +1153,38 @@ fn try_bob_accordant_lock_to_bob_accordant_lock_final(
...
-                    wallet,
-                },
+                BobAccordantLockFinal { wallet },
             )))
         }
         _ => handle_bob_swap_interrupt_after_lock(event, runtime),
     }
 }
 
 fn try_bob_accordant_lock_final_to_bob_buy_final(
```
[the remark this answers is not in the document]
Comment (TheCharlatan) [reply src/swapd/swap_state.rs]: ce0669d326d0cadb58996117a32061f217cee082

**Review by h4sh3d — APPROVED** [review]
tACK

**Pull request merged**